### ISSUE TITLE
Components won't build in Gatsby SSR

### ISSUE THREAD

**Issue by jlev** [trussworks/react-uswds#380]
**Describe the bug**
When using these components in Gatsby, they work in the develop stage but not when building for server-side rendering.

**To Reproduce**
1. Create new Gatsby site and `yarn add @trussworks/react-uswds`
2. Use any components in pages
3. `gatsby develop` works
4. `gatsby build` fails

**Expected behavior**
Initial issue is with use of `window` in index.js: 
```
ERROR #95312 
"window" is not available during server side rendering.
```

This can be fixed by using setting `gatsby-node.js` to use webpack null-loader for `/react-uswds/`, per instructions at https://www.gatsbyjs.org/docs/debugging-html-builds/#fixing-third-party-modules

Subsequent issue is harder to debug, errors when using any of these components and building.
```
  Error: Minified React error #130; visit https://reactjs.org/docs/error-decoder.html?invariant=130&args[]=undefined&args[]= for  the full message or use the non-minified dev environment for full errors and additional helpful warnings.  
```

Because there's not a lot of information in this error message, it's difficult to tell what's going on. And since there's no error when running develop, there's not a lot more to go on here.

**Additional context**
My example site in development is at https://github.com/spacedogXYZ/crush2020, if this is helpful to debug. Other React USWDS implementations do not appear to have this problem (eg, https://github.com/18F/federalist-uswds-gatsby) but this repo looks like it is currently under development and would be great if it worked for this use case.

**Environment:**
 - Mac OS Catalina 10.15.5
 - Node versions v13.8.0 and v12.18.3 (LTS)

**Comment by jlev** [comment]
@haworku let me know if you'd like me to do any other debugging. I'm not super familiar with typescript, but I can send along any compile logs that might be helpful...

**Comment by haworku** [comment]
Hey thanks @jlev for filing this issue and for this comment! The information you already provided is enough.   @ahobson is looking into things a bit

**Comment by ahobson** [comment]
Once we make a release, you should remove the `null-loader` config from `gatsby-node.js` (or remove the file entirely since it's not needed any more).

**Comment by jlev** [comment]
Thank you for the quick response! I've checked out the fix branch and built locally with `yarn build` and linked to my own package.json, and now the SSR with `gatsby build` works.

However, it now the regular `gatsby develop` does not, and has a lot of webpack errors and warnings for the generated lib...

I can wait for a real release to see if your release build is different, but at least from my end this doesn't appear to resolve the problem.

```
 ERROR #98123  WEBPACK

Generating development JavaScript bundle failed


/Users/jlev/Code/Spacedog/react-uswds/lib/index.js
  1:1      error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  1:128    error    'define' is not defined                                                no-undef
  1:139    error    'define' is not defined                                                no-undef
  1:410    warning  Unexpected use of comma operator                                       no-sequences
  1:440    warning  Unexpected use of comma operator                                       no-sequences
  1:543    error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  1:651    warning  Unexpected use of comma operator                                       no-sequences
  1:736    warning  Unexpected use of comma operator                                       no-sequences
  1:838    warning  Unexpected use of comma operator                                       no-sequences
  1:1101   warning  Unexpected use of comma operator                                       no-sequences
  6:3      error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:15     warning  'use strict' is unnecessary inside of modules                          strict
  6:342    error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:459    warning  'use strict' is unnecessary inside of modules                          strict
  6:472    error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:520    warning  Unexpected use of comma operator                                       no-sequences
  6:608    warning  'use strict' is unnecessary inside of modules                          strict
  6:621    error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:669    warning  Unexpected use of comma operator                                       no-sequences
  6:1537   error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:1617   warning  Unexpected use of comma operator                                       no-sequences
  6:2528   warning  'use strict' is unnecessary inside of modules                          strict
  6:2541   error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:2589   warning  Unexpected use of comma operator                                       no-sequences
  6:3509   error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:3601   warning  Unexpected use of comma operator                                       no-sequences
  6:4231   warning  'use strict' is unnecessary inside of modules                          strict
  6:4244   error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:4292   warning  Unexpected use of comma operator                                       no-sequences
  6:4753   warning  'use strict' is unnecessary inside of modules                          strict
  6:4766   error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:4814   warning  Unexpected use of comma operator                                       no-sequences
  6:5731   warning  'use strict' is unnecessary inside of modules                          strict
  6:5744   error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:5792   warning  Unexpected use of comma operator                                       no-sequences
  6:6753   error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:6828   warning  Unexpected use of comma operator                                       no-sequences
  6:7455   warning  'use strict' is unnecessary inside of modules                          strict
  6:7468   error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:7516   warning  Unexpected use of comma operator                                       no-sequences
  6:8694   warning  Unexpected use of comma operator                                       no-sequences
  6:9277   warning  Unexpected use of comma operator                                       no-sequences
  6:9390   warning  'use strict' is unnecessary inside of modules                          strict
  6:9403   error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:9451   warning  Unexpected use of comma operator                                       no-sequences
  6:9773   warning  'use strict' is unnecessary inside of modules                          strict
  6:9786   error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:9834   warning  Unexpected use of comma operator                                       no-sequences
  6:10776  warning  'use strict' is unnecessary inside of modules                          strict
  6:10789  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:10837  warning  Unexpected use of comma operator                                       no-sequences
  6:10928  warning  Unexpected mix of '||' and '&&'                                        no-mixed-operators
  6:10948  warning  Unexpected mix of '||' and '&&'                                        no-mixed-operators
  6:11264  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:11275  warning  Unexpected use of comma operator                                       no-sequences
  6:11413  warning  Missing '()' invoking a constructor                                    new-parens
  6:11432  warning  'u' is a function                                                      no-func-assign
  6:11454  warning  Unexpected use of comma operator                                       no-sequences
  6:11736  warning  Unexpected mix of '||' and '&&'                                        no-mixed-operators
  6:11748  warning  Unexpected mix of '||' and '&&'                                        no-mixed-operators
  6:11828  warning  Unexpected use of comma operator                                       no-sequences
  6:12190  warning  'e' was used before it was defined                                     no-use-before-define
  6:12193  warning  Unexpected use of comma operator                                       no-sequences
  6:12262  warning  Unexpected use of comma operator                                       no-sequences
  6:12742  warning  'use strict' is unnecessary inside of modules                          strict
  6:12755  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:12803  warning  Unexpected use of comma operator                                       no-sequences
  6:17868  warning  'use strict' is unnecessary inside of modules                          strict
  6:17881  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:17929  warning  Unexpected use of comma operator                                       no-sequences
  6:19513  warning  'use strict' is unnecessary inside of modules                          strict
  6:19526  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:19574  warning  Unexpected use of comma operator                                       no-sequences
  6:19679  warning  Unexpected mix of '||' and '&&'                                        no-mixed-operators
  6:19699  warning  Unexpected mix of '||' and '&&'                                        no-mixed-operators
  6:20015  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:20026  warning  Unexpected use of comma operator                                       no-sequences
  6:20157  warning  Missing '()' invoking a constructor                                    new-parens
  6:20176  warning  'l' is a function                                                      no-func-assign
  6:20198  warning  Unexpected use of comma operator                                       no-sequences
  6:21623  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:21629  warning  Unexpected use of comma operator                                       no-sequences
  6:21746  warning  'use strict' is unnecessary inside of modules                          strict
  6:21759  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:21807  warning  Unexpected use of comma operator                                       no-sequences
  6:21894  warning  Unexpected mix of '||' and '&&'                                        no-mixed-operators
  6:21914  warning  Unexpected mix of '||' and '&&'                                        no-mixed-operators
  6:22230  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:22241  warning  Unexpected use of comma operator                                       no-sequences
  6:22427  warning  Missing '()' invoking a constructor                                    new-parens
  6:22446  warning  'c' is a function                                                      no-func-assign
  6:22468  warning  Unexpected use of comma operator                                       no-sequences
  6:25507  warning  'use strict' is unnecessary inside of modules                          strict
  6:25520  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:25526  warning  Unexpected use of comma operator                                       no-sequences
  6:25815  warning  'use strict' is unnecessary inside of modules                          strict
  6:25828  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:25834  warning  Unexpected use of comma operator                                       no-sequences
  6:26959  warning  'use strict' is unnecessary inside of modules                          strict
  6:26972  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:26978  warning  Unexpected use of comma operator                                       no-sequences
  6:27995  warning  'use strict' is unnecessary inside of modules                          strict
  6:28008  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:28056  warning  Unexpected use of comma operator                                       no-sequences
  6:28268  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:28280  warning  Unexpected use of comma operator                                       no-sequences
  6:29165  warning  'use strict' is unnecessary inside of modules                          strict
  6:29178  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:29226  warning  Unexpected use of comma operator                                       no-sequences
  6:29860  warning  'use strict' is unnecessary inside of modules                          strict
  6:29873  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:29921  warning  Unexpected use of comma operator                                       no-sequences
  6:30866  warning  'use strict' is unnecessary inside of modules                          strict
  6:30879  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:30927  warning  Unexpected use of comma operator                                       no-sequences
  6:31356  warning  'use strict' is unnecessary inside of modules                          strict
  6:31369  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:31417  warning  Unexpected use of comma operator                                       no-sequences
  6:32367  warning  'use strict' is unnecessary inside of modules                          strict
  6:32380  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:32428  warning  Unexpected use of comma operator                                       no-sequences
  6:32622  warning  'use strict' is unnecessary inside of modules                          strict
  6:32635  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:32683  warning  Unexpected use of comma operator                                       no-sequences
  6:33800  warning  'use strict' is unnecessary inside of modules                          strict
  6:33813  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:33861  warning  Unexpected use of comma operator                                       no-sequences
  6:34745  warning  Expected '===' and instead saw '=='                                    eqeqeq
  6:34776  warning  Expected '===' and instead saw '=='                                    eqeqeq
  6:34809  warning  Expected '===' and instead saw '=='                                    eqeqeq
  6:35155  warning  'use strict' is unnecessary inside of modules                          strict
  6:35168  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:35216  warning  Unexpected use of comma operator                                       no-sequences
  6:36135  warning  'use strict' is unnecessary inside of modules                          strict
  6:36148  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:36196  warning  Unexpected use of comma operator                                       no-sequences
  6:37160  warning  'use strict' is unnecessary inside of modules                          strict
  6:37173  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:37221  warning  Unexpected use of comma operator                                       no-sequences
  6:37581  warning  'use strict' is unnecessary inside of modules                          strict
  6:37594  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:37642  warning  Unexpected use of comma operator                                       no-sequences
  6:38055  warning  'use strict' is unnecessary inside of modules                          strict
  6:38068  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:38116  warning  Unexpected use of comma operator                                       no-sequences
  6:39212  warning  'use strict' is unnecessary inside of modules                          strict
  6:39225  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:39273  warning  Unexpected use of comma operator                                       no-sequences
  6:40183  warning  'use strict' is unnecessary inside of modules                          strict
  6:40196  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:40244  warning  Unexpected use of comma operator                                       no-sequences
  6:41298  warning  'use strict' is unnecessary inside of modules                          strict
  6:41311  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:41359  warning  Unexpected use of comma operator                                       no-sequences
  6:42285  warning  'use strict' is unnecessary inside of modules                          strict
  6:42298  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:42346  warning  Unexpected use of comma operator                                       no-sequences
  6:43289  warning  'use strict' is unnecessary inside of modules                          strict
  6:43302  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:43350  warning  Unexpected use of comma operator                                       no-sequences
  6:44712  warning  'use strict' is unnecessary inside of modules                          strict
  6:44725  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:44731  warning  Unexpected use of comma operator                                       no-sequences
  6:45448  warning  'use strict' is unnecessary inside of modules                          strict
  6:45461  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:45509  warning  Unexpected use of comma operator                                       no-sequences
  6:46521  warning  'use strict' is unnecessary inside of modules                          strict
  6:46534  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:46582  warning  Unexpected use of comma operator                                       no-sequences
  6:47674  warning  'use strict' is unnecessary inside of modules                          strict
  6:47687  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:47735  warning  Unexpected use of comma operator                                       no-sequences
  6:48548  warning  'use strict' is unnecessary inside of modules                          strict
  6:48561  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:48609  warning  Unexpected use of comma operator                                       no-sequences
  6:49451  warning  'use strict' is unnecessary inside of modules                          strict
  6:49464  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:49512  warning  Unexpected use of comma operator                                       no-sequences
  6:50682  warning  'use strict' is unnecessary inside of modules                          strict
  6:50695  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:50743  warning  Unexpected use of comma operator                                       no-sequences
  6:51851  warning  'use strict' is unnecessary inside of modules                          strict
  6:51864  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:51912  warning  Unexpected use of comma operator                                       no-sequences
  6:52829  warning  'use strict' is unnecessary inside of modules                          strict
  6:52842  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:52890  warning  Unexpected use of comma operator                                       no-sequences
  6:53063  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:53141  warning  Unexpected use of comma operator                                       no-sequences
  6:53977  warning  'use strict' is unnecessary inside of modules                          strict
  6:53990  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:54038  warning  Unexpected use of comma operator                                       no-sequences
  6:54858  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:54933  warning  Unexpected use of comma operator                                       no-sequences
  6:55552  warning  'use strict' is unnecessary inside of modules                          strict
  6:55565  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:55613  warning  Unexpected use of comma operator                                       no-sequences
  6:56483  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:56561  warning  Unexpected use of comma operator                                       no-sequences
  6:57264  warning  'use strict' is unnecessary inside of modules                          strict
  6:57277  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:57325  warning  Unexpected use of comma operator                                       no-sequences
  6:57896  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:57974  warning  Unexpected use of comma operator                                       no-sequences
  6:58627  warning  'use strict' is unnecessary inside of modules                          strict
  6:58640  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:58688  warning  Unexpected use of comma operator                                       no-sequences
  6:59021  warning  'use strict' is unnecessary inside of modules                          strict
  6:59034  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:59082  warning  Unexpected use of comma operator                                       no-sequences
  6:59979  warning  'use strict' is unnecessary inside of modules                          strict
  6:59992  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:60040  warning  Unexpected use of comma operator                                       no-sequences
  6:61368  warning  'use strict' is unnecessary inside of modules                          strict
  6:61381  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:61429  warning  Unexpected use of comma operator                                       no-sequences
  6:62388  warning  'use strict' is unnecessary inside of modules                          strict
  6:62401  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:62449  warning  Unexpected use of comma operator                                       no-sequences
  6:63553  warning  'use strict' is unnecessary inside of modules                          strict
  6:63566  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:63614  warning  Unexpected use of comma operator                                       no-sequences
  6:64560  warning  'use strict' is unnecessary inside of modules                          strict
  6:64573  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:64621  warning  Unexpected use of comma operator                                       no-sequences
  6:65577  warning  'use strict' is unnecessary inside of modules                          strict
  6:65590  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:65638  warning  Unexpected use of comma operator                                       no-sequences
  6:66588  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:66663  warning  Unexpected use of comma operator                                       no-sequences
  6:67263  warning  'use strict' is unnecessary inside of modules                          strict
  6:67276  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:67324  warning  Unexpected use of comma operator                                       no-sequences
  6:67450  warning  Unexpected mix of '||' and '&&'                                        no-mixed-operators
  6:67470  warning  Unexpected mix of '||' and '&&'                                        no-mixed-operators
  6:67786  error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
  6:67797  warning  Unexpected use of comma operator                                       no-sequences
  6:67961  warning  Missing '()' invoking a constructor                                    new-parens
  6:67980  warning  'u' is a function                                                      no-func-assign
  6:68002  warning  Unexpected use of comma operator                                       no-sequences

✖ 240 problems (76 errors, 164 warnings)
  0 errors and 61 warnings potentially fixable with the `--fix` option.

```

**Comment by ahobson** [comment]
I'm not positive what the problem is with `yarn link`, but I speculate it's trying to compile the source to `react-uswds` which is in TypeScript.

If I simulate installing the new code, everything seems to work.

```
$ cd react-uswds
$ yarn build
$ cd ../crush2020
$ rsync -av ../react-uswds/lib/* node_modules/@trussworks/react-uswds/lib/
## this works fine
$ ./node_modules/.bin/gatsby develop 
## this works fine
$ ./node_modules/.bin/gatsby build
```

**Comment by jlev** [comment]
Yeah, it might be due to the typescript compilation step. I told eslint to ignore that file, and the warnings were suppressed.

I may need to wait for a new release before deploying, because netlify does all the build steps for me, but this definitely helps me keep moving on this project.

**Comment by haworku** [comment]
This change is available on npm in version `1.9.0`.
